src/storj.c

```diff
@@ -415,18 +349,68 @@ static void get_file_info_request_worker(uv_work_t *work)
     const char *encrypted_file_name = json_object_get_string(filename);
     if (encrypted_file_name) {
         char *decrypted_file_name;
-        int error_status = decrypt_meta(encrypted_file_name, key,
-                                        &decrypted_file_name);
+        int error_status = decrypt_file_name(req->encrypt_options->mnemonic,
+                                             req->bucket_id,
+                                             encrypted_file_name,
+                                             &decrypted_file_name);
         if (!error_status) {
             req->file->decrypted = true;
             req->file->filename = decrypted_file_name;
-        } else {
+        } else if (error_status == STORJ_META_DECRYPTION_ERROR) {
         	req->file->decrypted = false;
         	req->file->filename = strdup(encrypted_file_name);
+        } else {
+            req->error_code = STORJ_MEMORY_ERROR;
         }
     }
 }
 
+static void get_file_id_request_worker(uv_work_t *work)
+{
+    get_file_id_request_t *req = work->data;
+    int status_code = 0;
+
+    char *encrypted_file_name;
+    if (encrypt_file_name(req->encrypt_options->mnemonic,
+                          req->bucket_id,
+                          req->file_name,
+                          &encrypted_file_name)) {
+        req->error_code = STORJ_MEMORY_ERROR;
+        return;
+    }
+
+    char *escaped_encrypted_file_name = str_replace("/", "%2F", encrypted_file_name);
+    if (!escaped_encrypted_file_name) {
+        req->error_code = STORJ_MEMORY_ERROR;
+        return;
+    }
+
+    free(encrypted_file_name);
+
+    char *path = str_concat_many(4, "/buckets/", req->bucket_id,
+                                    "/file-ids/", escaped_encrypted_file_name);
+    if (!path) {
+        req->error_code = STORJ_MEMORY_ERROR;
+        return;
+    }
+
+    free(escaped_encrypted_file_name);
```

Comment:
This may not get freed, could use `goto` with cleanup label

Reply:
I pushed https://github.com/Storj/libstorj/pull/414/commits/5c4069327c0a30a53317c19f5515c3d71bc7f0c2 that does this.